interfaces/cython/cantera/thermo.pyx

```diff
@@ -809,15 +809,19 @@ cdef class ThermoPhase(_SolutionBase):
         self.state = original_state
         return arr
 
-    def set_equivalence_ratio(self, phi, fuel, oxidizer, basis='mole'):
+    def set_equivalence_ratio(self, phi, fuel, oxidizer, basis="mole", *, diluent=None, fraction=None):
         """
         Set the composition to a mixture of ``fuel`` and ``oxidizer`` at the
         specified equivalence ratio ``phi``, holding temperature and pressure
         constant. Considers the oxidation of C to CO2, H to H2O and S to SO2.
         Other elements are assumed not to participate in oxidation (that is,
         N ends up as N2). The ``basis`` determines the fuel and oxidizer
         compositions: ``basis='mole'`` means mole fractions (default),
-        ``basis='mass'`` means mass fractions::
+        ``basis='mass'`` means mass fractions. After a mixture given a
+        ``phi`` from ``fuel`` and ``oxidizer`` is  created, it can further
+        be diluted by a ``diluent`` based on a mixing ``fraction``. `See also
```

Comment:
Hmm. I’m not sure that I am following completely: as long as an inert mixture is used, the equivalence ratio should not be affected by a diluent (a user could certainly wreak havoc by specifying reactants as diluents, but that is not something that needs to be handled as this would be imho a ‘user error’). To my understanding, `fraction` just specifies how the diluent ratio is specified: either as a fraction of “fuel, oxidizer, or the [reactant] mixture”. If I’m not mistaken, this is how your tests are already implemented?

Reply:
Maybe we mean the same thing. But even if the diluent is just another fuel species, everything is well defined.
Lets make an example:
Assume I want a mixture of H2 and O2 at stoichiometric conditions. So I would write:
set_equivalence_ratio(1, "H2", "O2")
So I get a mixture with `X_H2 = 2 / 3` and` X_O2 = 1 / 3` or `X_H2 / X_O2 = 2`.
Now I want to dilute this mixture with "CH4" in a way that the fraction of fuel (i.e. H2) in the final mixture is 0.1.
For this, I would write:
set_equivalece_ratio(1, "H2", "O2", fraction={"fuel": 0.1}, diluent="CH4")
This gives a mixture with `X_H2 = 0.1, X_O2 = 0.05` and `X_CO2 = 0.85`.
Whatever this composition is, it is always guaranteed that X_H2 is 0.1 (i.e. according to `fraction={"fuel": 0.1}`) and X_H2/X_O2 is guaranteed to be 2 (i.e. `phi=2`).
The goal is of the function with dilution is not to get a final mixture with the specified `phi`, but to dilute a oxidizer/fuel mixture at a specified phi with an arbitrary diluent.

Reply:
> Maybe we mean the same thing.

I think so. The way it works is that the equivalence ratio is defined by fuel and oxidizer, whereas the diluent is not considered. The fraction just specifies how the diluent is quantified.

Reply:
OK, nice 👍  Then I will prepare the next commit

Reply:
I don't think that the docstring accurately describes the dilution option yet. Please address the section "specifying
the amount of diluent, fuel or oxidizer in the mixture." ... perhaps "[...]. The amount of diluent is quantified as a fraction of fuel, oxidizer or the fuel/oxidizer mixture." ?